### DISCUSSION TITLE
How to use the loggerLink in a vanilla trpc client

### DISCUSSION THREAD

**Discussion by seansu4you87** [trpc/trpc#1837] [Q&A] [answered]
Hi there, are there any examples or instructions for how to use the `loggerLink` (and other links) in a vanilla trpc client?  I found the example in https://github.com/trpc/trpc/blob/main/examples/next-prisma-starter/src/pages/_app.tsx#L59 but that references trpc next helpers

Thanks

**Comment by KATT** [comment]
https://trpc.io/docs/links, is this enough?

**Comment by seansu4you87** [comment]
Those use `withTRPC` and other helper methods, whereas I'm just trying to configure a vanilla client with `createTRPCClient`

**Comment by seansu4you87** [comment, marked as answer]
In case it's helpful I'm configuring my client like so:

```
export const trpc = createTRPCClient<AppRouter>({
  url: DEVELOPMENT_URL,
  links: [
    loggerLink({
      enabled: (opts) => true,
    }),
  ],
});
```

And it's logging things out, but I'm getting an error: `TypeError: link is not a function`

![Screen Shot 2022-04-27 at 11 51 39 AM](https://user-images.githubusercontent.com/507797/165598699-6bb01ae5-1d41-4ff2-96ee-2bbb602f06dc.png)

**Reply by KATT** [reply]
You need to add a httpBatchLink at the end

**Reply by KATT** [reply]
The url and links shouldn't be used together, only supplying a url is a shortcut for some links

**Reply by seansu4you87** [reply]
Ahh, I see that fixed my errors.  Thanks!

Does that mean that there has to be an `httpBatchLink` in all clients, and that all trpc requests are batched by default?
